code/modules/fish/fishtank.dm

```diff
@@ -313,14 +313,17 @@
 	egg_list.Cut()									//Destroy any excess eggs, clearing the egg_list
 
 /obj/machinery/fishtank/proc/harvest_fish(var/mob/user)
-	if(!fish_count)									//Can't catch non-existant fish!
+	if(fish_count <= 0)									//Can't catch non-existant fish!	
 		to_chat(usr, "There are no fish in \the [src] to catch!")
 		return
 	var/list/fish_names_list = list()
 	for(var/datum/fish/fish_type in fish_list)
 		fish_names_list += list("[fish_type.fish_name]" = fish_type)
 	var/caught_fish = input("Select a fish to catch.", "Fishing") as null|anything in fish_names_list		//Select a fish from the tank
-	if(caught_fish)
+	if(fish_count <= 0)
```

Comment:
As above, this shouldn't be possible to go negative, but will still evaluate as true when fish_count is 0 so it works like this or as `if(!fish_count)`